utils.js

```diff
@@ -0,0 +1,35 @@
+class RecentlyUsedList {
+
+    constructor() {
+        this.items = [];
+    }
+
+    add(newItem) {
+        if (this.items.indexOf(newItem)) {
```

Comment:
While this is not wrong, but better to consider using new Set() to handle uniqueness of the values in an array instead of using three different functions: indexOf/Splice/Unshift.